src/werkzeug/routing/rules.py

```diff
@@ -625,7 +625,7 @@ def _parse_rule(self, rule: str) -> t.Iterable[RulePart]:
                     -len(argument_weights),
                     argument_weights,
                 )
-                if final:
+                if not static:
                     content += r"$\Z"
```

Comment:
Shouldn't this just be `\Z`? The `$` shouldn't be needed.

Reply:
I think this is the reason, https://github.com/pallets/werkzeug/commit/d761d0e7acceb89f56d76434a74aeacf95d5ed49 looks like you know better :)